### PR TITLE
Add version note for `has` property

### PR DESCRIPTION
**Pull request by tubbo**
The guides for `next.config.js` never specify that the `has: { }` syntax is only available for v10.1, and not for v10.0. Add a note that indicates these features are only available in the latest version.

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

**Comment by leerob** [comment]
There's merge conflicts because the docs have been updated to remove the experimental note: https://github.com/vercel/next.js/blob/canary/docs/api-reference/next.config.js/rewrites.md#header-cookie-and-query-matching

Instead, let's add this to denote the version it was added under that section

```
<details>
  <summary><b>Version History</b></summary>

| Version   | Changes                   |
| --------- | ------------------- |
| `v10.0.2` | `has` added.           |

</details>
```

**Comment by tubbo** [comment]
@leerob done!

File: docs/api-reference/next.config.js/rewrites.md [hunk 1]
```diff
@@ -13,6 +13,15 @@ description: Add rewrites to your Next.js app.
   </ul>
 </details>
 
+<details>
```
Comment (leerob) [inline docs/api-reference/next.config.js/rewrites.md:16]: Could you add this to `headers.md` too? Then good to go!

File: docs/api-reference/next.config.js/redirects.md [hunk 2]
```diff
@@ -13,6 +13,15 @@ description: Add redirects to your Next.js app.
   </ul>
 </details>
 
+<details>
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes      |
+| --------- | ------------ |
+| `v10.1.0` | `has` added. |
```
Comment (leerob) [inline docs/api-reference/next.config.js/redirects.md:21]: 10.2 is when it was stable. Let's use that.

Comment (leerob) [reply docs/api-reference/next.config.js/redirects.md:21]: https://nextjs.org/10-2

Comment (timneutkens) [reply docs/api-reference/next.config.js/redirects.md:21]:
```suggestion
| `v10.2.0` | `has` added. |
```

File: docs/api-reference/next.config.js/rewrites.md [hunk 3]
```diff
@@ -13,6 +13,15 @@ description: Add rewrites to your Next.js app.
...
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes      |
+| --------- | ------------ |
+| `v10.1.0` | `has` added. |
```
Comment (timneutkens) [inline docs/api-reference/next.config.js/rewrites.md:21]:
```suggestion
| `v10.2.0` | `has` added. |
```

File: docs/api-reference/next.config.js/rewrites.md (hunk 1 above)
Comment (timneutkens) [reply docs/api-reference/next.config.js/rewrites.md:16]: I've added it to headers as well 👍

**Review by timneutkens — APPROVED** [review]
Looks great! Thanks.

**Pull request merged**